torch/jit/__init__.py

```diff
@@ -1526,21 +1541,28 @@ def __init__(self, original, stubs):
             # __init__ can run correctly
```

Comment:
We need a plan to get rid of WeakScriptModuleProxy. Eventually all ScriptModules will just always be this class, and they will not longer be proxies at all. It is difficult to understand all the components we have after this patch, so we need some followup effort merging the concepts.

Reply:
Most of the functionality here can operate directly on a ScriptModule instance instead of constructing a `WeakScriptModuleProxy`, so deleting it should be pretty painless. There is still the question of what something like this would do:

```python
scripted_linear = torch.jit.script(nn.Linear(10, 10))
isinstance(scripted_linear, nn.Linear) # true or false?
```

Intuitively I feel like it should be `False` since it's constructing a new module, but I can see people doing something like this, which could lead to confusing behavior.

```python
class A(nn.Module):
def __init__(self, mod):
	self.module = mod
	if isinstance(self.module, nn.Linear):
		...

A(torch.jit.script(B())
```